dashboard/app/controllers/activities_controller.rb

```diff
@@ -31,7 +31,9 @@ def milestone
     # disabled. (A cached view might post to this action even if milestone posts
     # are disabled in the gatekeeper.)
     enabled = Gatekeeper.allows('postMilestone', where: {script_name: script_name}, default: true)
-    unless enabled
+    # Exception: Always post milestone if we passed the last level of the script.
+    # Keep this logic in sync with code-studio/reporting#sendReport on the client.
+    unless enabled || (solved && @script_level.try(:final_level?))
```

Comment:
Like the idea of "mostly disable" and "always disable" settings. Thanks for adding.